dpnp/backend/custom_kernels_statistics.cpp

```diff
@@ -139,3 +139,146 @@ void custom_cov_c(void* array1_in, void* result1, size_t nrows, size_t ncols)
 }
 
 template void custom_cov_c<double>(void* array1_in, void* result1, size_t nrows, size_t ncols);
+
+template <typename _DataType>
+class custom_max_c_kernel;
+
+template <typename _DataType>
+void custom_max_c(void* array1_in, void* result1, size_t* shape, size_t ndim, size_t* axis, size_t naxis)
+{
+    _DataType* array_1 = reinterpret_cast<_DataType*>(array1_in);
+    _DataType* result = reinterpret_cast<_DataType*>(result1);
+
+    size_t size = 1;
+    for (size_t i = 0; i < ndim; ++i)
+    {
+        size *= shape[i];
+    }
+
+    auto policy = oneapi::dpl::execution::make_device_policy<class custom_max_c_kernel<_DataType>>(DPNP_QUEUE);
+
+    _DataType* res = std::max_element(policy, array_1, array_1 + size);
+    policy.queue().wait();
+
+    result[0] = *res;
+
+#if 0
+    std::cout << "max result " << result[0] << "\n";
+#endif
+}
+
+template void
+    custom_max_c<double>(void* array1_in, void* result1, size_t* shape, size_t ndim, size_t* axis, size_t naxis);
+template void
+    custom_max_c<float>(void* array1_in, void* result1, size_t* shape, size_t ndim, size_t* axis, size_t naxis);
+template void
+    custom_max_c<long>(void* array1_in, void* result1, size_t* shape, size_t ndim, size_t* axis, size_t naxis);
+template void custom_max_c<int>(void* array1_in, void* result1, size_t* shape, size_t ndim, size_t* axis, size_t naxis);
+
+template <typename _DataType, typename _ResultType>
+void custom_mean_c(void* array1_in, void* result1, size_t* shape, size_t ndim, size_t* axis, size_t naxis)
+{
+    _ResultType* result = reinterpret_cast<_ResultType*>(result1);
+
+    size_t size = 1;
+    for (size_t i = 0; i < ndim; ++i)
+    {
+        size *= shape[i];
+    }
+
+    _DataType* sum = reinterpret_cast<_DataType*>(dpnp_memory_alloc_c(1 * sizeof(_DataType)));
+
+    custom_sum_c<_DataType>(array1_in, sum, size);
```

Comment:
```suggestion
    _DataType sum;

    custom_sum_c<_DataType>(array1_in, &sum, size);
```

Reply:
oh... no. we need USM memory for kernel. sorry - my proposal is wrong.
perhaps it is better to use `*result` memory for this